packages/near-membrane-base/src/__tests__/shared.spec.ts

```diff
@@ -0,0 +1,45 @@
+import { ObjectLookupOwnGetter, WeakMapHas } from '../shared';
+
+describe('ObjectLookupOwnGetter', () => {
+    it('should return undefined when object argument is null', () => {
+        expect.assertions(1);
+        expect(ObjectLookupOwnGetter(null)).toBe(undefined);
+    });
+    it('should return undefined when object argument is undefined', () => {
+        expect.assertions(1);
+        expect(ObjectLookupOwnGetter(undefined)).toBe(undefined);
+    });
+    it('should return undefined when object argument has no such property', () => {
+        expect.assertions(1);
+        expect(ObjectLookupOwnGetter({}, 'foo')).toBe(undefined);
+    });
+    it('should return getter when object argument has property and getter', () => {
+        expect.assertions(1);
+        const o = {
+            get x() {
+                return 1;
+            },
+        };
+        expect(ObjectLookupOwnGetter(o, 'x')).toBe(Object.getOwnPropertyDescriptor(o, 'x').get);
+    });
```

Comment:
☝️ Add a check that it doesn't lookup inherited getters. The real unguarded `__lookupGetter__` will lookup inherited ones (which is why we added a hasOwnProperty check to the helper).